src/appleseed/renderer/modeling/bsdf/glassbsdf.cpp

```diff
@@ -261,6 +301,14 @@ namespace
                         wi,
                         sample.m_value.m_glossy,
                         sample.m_probability);
+
+                    add_energy_compensation_term(
+                        mdf,
+                        values,
+                        Vector3f(0.0f, 1.0f, 0.0f),
```

Comment:
Why the vertical normal here? In what space is the normal expressed?

Reply:
wi and wo are both in local (shading basis) space.